federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -0,0 +1,454 @@
+		frs, err = frsc.fedClient.Extensions().ReplicaSets(frs.Namespace).Update(frs)
+		if err != nil {
+			return err
+		}
+	}
+
+	var totalReplicas int64 = 0
+	for _, replicas := range scheduleResult {
+		totalReplicas += replicas
+	}
+	if int32(totalReplicas) != *frs.Spec.Replicas {
+		// replicas don't match, re-schedule
+		clusters, err := frsc.fedInformer.GetReadyClusters()
+		if err != nil {
+			return err
+		}
+		scheduleResult = frsc.schedule(*frs.Spec.Replicas, clusters, scheduleResult, nil)
+	}
+
+	glog.Infof("Start syncing local replicaset %v", scheduleResult)
+
+	fedStatus := extensionsv1.ReplicaSetStatus{ObservedGeneration: frs.Generation}
+	for clusterName, replicas := range scheduleResult {
+		// TODO: updater or parallelizer doesnn't help as results are needed for updating fed rs status
+		clusterClient, err := frsc.fedInformer.GetClientsetForCluster(clusterName)
+		if err != nil {
+			return err
+		}
+		var lrs *extensionsv1.ReplicaSet
+		lrsObj, exists, err := frsc.fedInformer.GetTargetStore().GetByKey(clusterName, key)
+		if err != nil {
+			return err
+		} else if !exists {
+			if replicas > 0 {
+				lrsClone, _ := api.Scheme.DeepCopy(frs)
+				lrs = lrsClone.(*extensionsv1.ReplicaSet)
+				lrs.ObjectMeta = apiv1.ObjectMeta{
+					Name:        lrs.Name,
+					Namespace:   lrs.Namespace,
+					Labels:      lrs.Labels,
+					Annotations: lrs.Annotations,
+				}
+				specReplicas := int32(replicas)
+				lrs.Spec.Replicas = &specReplicas
+				lrs, err = clusterClient.Extensions().ReplicaSets(frs.Namespace).Create(lrs)
+				if err != nil {
+					return err
+				}
+				fedStatus.Replicas += lrs.Status.Replicas
+				fedStatus.FullyLabeledReplicas += lrs.Status.FullyLabeledReplicas
+			}
+		} else {
+			lrs = lrsObj.(*extensionsv1.ReplicaSet)
+			lrsExpectedSpec := frs.Spec
+			specReplicas := int32(replicas)
+			lrsExpectedSpec.Replicas = &specReplicas
+			if !reflect.DeepEqual(lrs.Spec, lrsExpectedSpec) {
+				lrs.Spec = lrsExpectedSpec
+				lrs, err = clusterClient.Extensions().ReplicaSets(frs.Namespace).Update(lrs)
+				if err != nil {
+					return err
+				}
+			}
+			fedStatus.Replicas += lrs.Status.Replicas
+			fedStatus.FullyLabeledReplicas += lrs.Status.FullyLabeledReplicas
+			if replicas == 0 {
+				err := clusterClient.Extensions().ReplicaSets(frs.Namespace).Delete(frs.Name, &api.DeleteOptions{})
+				if err != nil && !errors.IsNotFound(err) {
+					return err
+				}
+			}
+		}
+	}
+	if fedStatus.Replicas != frs.Status.Replicas || fedStatus.FullyLabeledReplicas != frs.Status.FullyLabeledReplicas {
+		frs.Status = fedStatus
+		_, err = frsc.fedClient.Extensions().ReplicaSets(frs.Namespace).UpdateStatus(frs)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (frsc *ReplicaSetController) deleteReplicaSet(key string) error {
+	glog.Infof("deleting replicaset: %v", key)
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	// try delete from all clusters
```

Comment:
Nothing. I'm not super happy with this situation either but @quinton-hoole wants to temporarily keep it this way until he finds a consistent and user-friendly way to execute deletes in federation.